bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -906,10 +907,19 @@ public Object evaluate(String script) throws SWTException {
 	// Feature in WebView2. ExecuteScript works regardless of IsScriptEnabled setting.
 	// Disallow programmatic execution manually.
 	if (!jsEnabled) return null;
-
+	if(inCallback) {
+		execute(script);
+		return null;
+	}
+	inEvaluate = true;
 	String script2 = "(function() {try { " + script + " } catch (e) { return '" + ERROR_ID + "' + e.message; } })();\0";
 	String[] pJson = new String[1];
-	int hr = callAndWait(pJson, completion -> webViewProvider.getWebView(true).ExecuteScript(script2.toCharArray(), completion));
+	int hr;
+	try {
+		hr = callAndWait(pJson, completion -> webViewProvider.getWebView(true).ExecuteScript(script2.toCharArray(), completion));
+	} finally {
+		inEvaluate = false;
+	}
 	if (hr != COM.S_OK) error(SWT.ERROR_FAILED_EVALUATE, hr);
```

Comment:
Since you don't need `hr` outside, you can put everything inside the `try` block
```suggestion
	try {
		int hr = callAndWait(pJson, completion -> webViewProvider.getWebView(true).ExecuteScript(script2.toCharArray(), completion));
	if (hr != COM.S_OK) error(SWT.ERROR_FAILED_EVALUATE, hr);
	} finally {
		inEvaluate = false;
	}
```

Reply:
Done in ce625749c181e36ff209703684cf01e9695a917d

Reply:
@amartya4256 please restore my changes from ce625749c181e36ff209703684cf01e9695a917d